File: frontend/src/types/TimelineEntry.ts
export interface TimelineEntry {
  id: number;
  userId: number;
  title: string;
  description: string;
  eventDate: string;
  entryType: "Achievement" | "Activity" | "Milestone" | "Memory";
  category: string;
  imageUrl: string;
  externalUrl: string;
  sourceApi: string;
  externalId: string;
  metadata: string;
  createdAt: string;
  updatedAt: string;
  user?: null;
}

export interface CreateTimelineEntry {
  userId: number;
  title: string;
  description: string;
  eventDate: string;
  entryType: "Achievement" | "Activity" | "Milestone" | "Memory";
  category: string;
  imageUrl?: string;
  externalUrl?: string;
}

export interface TimelineStats {
  totalEntries: number;
  entriesByType: Array<{ entryType: string; count: number }>;
  entriesByCategory: Array<{ category: string; count: number }>;
}

export interface TimelineFilter {
  query?: string;
  category?: string;
  entryType?: string;
  userId: number;
}

File: backend/PersonalTimelineAPI/Models/TimelineEntry.cs
namespace PersonalTimelineAPI.Models
{
    public class TimelineEntry
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public DateTime EventDate { get; set; }
        public string EntryType { get; set; } = "Activity"; 
        public string Category { get; set; } = "General";
        public string ImageUrl { get; set; } = string.Empty;
        public string ExternalUrl { get; set; } = string.Empty;
        public string SourceApi { get; set; } = "Manual"; 
        public string ExternalId { get; set; } = string.Empty; 
        public string Metadata { get; set; } = "{}"; 
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        
        // Navigation property
        public User? User { get; set; }
    }
}